server/routes/hdc.js

```diff
@@ -322,3 +379,15 @@ module.exports = function({logger, licenceService, conditionsService, prisonerSe
     return router;
 };
 
+function auditableData(input) {
+    return Object.keys(input).reduce((objectBuilt, key) => {
+        if (notAcceptedKeys.includes(key)) {
+            return {...objectBuilt};
+        } else {
+            const value = input[key] || '';
+            return {...objectBuilt, ...{[key]: value}};
+        }
+    }, {});
+}
```

Comment:
There is a nice Ramda function https://ramdajs.com/docs/#omit that could be used to filter an object. Would look like `R.omit(['nomisId', '_csrf', 'anchor'], input)`